ui_testing/testcases/basic_tests/test06_orders.py

```diff
@@ -1890,3 +1892,111 @@ def test028_update_test_unit_with_delete_in_form(self):
         testunit_name = row_with_headers['Test Unit']
         self.base_selenium.LOGGER.info(" + Test unit : {}".format(testunit_name))
         self.assertIn(testunit_name, testunit_name)
+
+
+    @parameterized.expand(['save_btn', 'cancel'])
+    def test029_update_article(self, save):
+        """
+        New: Orders: Edit Approach: I can update the article successfully and press on ok button
+        then press on cancel button, Nothing updated
+
+        New: Orders: Edit Approach: I can update the article filed successfully with save button
+
+        LIMS-3423
+        LIMS-4297
+        :return:
+        """
+
+        """
+        test case logic
+        i'll filter with raw material material type just to make sure that it has multiple articles
+        and then select and record
+        and then update it's article and try to press save and press cancel
+        """
+
+        self.base_selenium.LOGGER.info('Filter with Raw Material material type, just to make sure that it has multiple articles')
+        rows=self.order_page.search(value='Raw Material')
+
+        self.base_selenium.LOGGER.info('select the required order to update it')
+        row = self.orders_page.get_random_order()
+        order_page_url = self.base_selenium.get_url()
+        self.base_selenium.click(element='order:suborder_table')
+
+        initial_article = self.order_page.get_article()
+        initial_test_plan = self.order_page.get_test_plan()
+        initial_test_unit = self.order_page.get_test_unit()
+
+
+        # Update Article
+        self.order_page.set_article(
+            article='r')
+        new_article = self.order_page.get_article()
+
+        # Update Test Plan
+        if self.order_page.get_test_plan():
```

Comment:
Once you update the article and press on ok button, the test plan will clear so why you want to clear it in your code so you need to set the test plan only to put another value